Added an attention mechanism to custom_unet and custom_vnet, a 3D U-net

moritzknolle opened the pull request:
- added an attention mechanism to custom_unet
- added custom_vnet a U-net performing 3D convolutions
- added the attention mechanism to custom_vnet as well

moritzknolle committed:
Added attention mechanisms for custom_unet

moritzknolle committed:
Added custom_vnet, a U-net performing 3D Convolutions

custom_vnet also supports an attention mechanism

moritzknolle committed:
reverted small change

karolzak commented:
Thanks @moritzknolle for your contribution!
I will resolve the conflict (as I caused it) and refactor the code a bit (move some stuff to common.py to avoid code duplicates, etc.) and then merge this PR to master.
Cheers!
KZ

karolzak committed:
upd custom_vnet.py

Modified docstring to refer and point to V-Net instead of UNet.
Changed `custom_unet` function name to `custom_vnet`

karolzak reviewed (approved):
approved with small fixes in docstrings + function name

karolzak committed:
Merge branch 'master' into development

karolzak committed:
upd custom_vnet.py to work with TF2.0

Updated imports to work for both "old" (standalone) and "new" (part of Tensorflow) Keras

karolzak reviewed (approved):
modified imports to support both old and new versions of Keras

karolzak committed:
Merge branch 'master' into development
